pkg/loki/loki.go

```diff
@@ -0,0 +1,89 @@
+package loki
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/loki/pkg/logcli/client"
+	"github.com/grafana/loki/pkg/loghttp"
+	"github.com/grafana/loki/pkg/logproto"
+	"github.com/pingcap/errors"
+)
+
+const (
+	equalMatcher    = "|="
+	regexMatcher    = "|~"
+)
+
+type lokiClient struct {
+	cli       *client.Client
+	startTime time.Time
+}
+
+// NewLokiClient creates a client to query loki.
+func NewLokiClient(address, username, password string) *lokiClient {
+	return &lokiClient{
+		cli: &client.Client{
+			Address:  address,
+			Username: username,
+			Password: password,
+		},
+		startTime: time.Now(),
```

Comment:
Seems the client will only fetch logs after `startTime`, if there are some logs being collected into Loki before the client init, we can't access them.

Reply:
Yes. So the loki client needs to be initialized first.

Reply:
If we want get all logs from loki, why not just set the start time to 0?

Reply:
No. I want to avoid the case that there already some history logs in loki with the same namespace. So just query from the time that the loki client initialized.

Reply:
But maybe we should use a better way to do this. Maybe add another label to do filtering.

Reply:
Get it, we can keep this solution by now.